apps/s_client.c

```diff
@@ -3192,8 +3117,8 @@ int s_client_main(int argc, char **argv)
     timeout.tv_usec = 500000;  /* some extreme round-trip */
     do {
         FD_ZERO(&readfds);
-        openssl_fdset(s, &readfds);
-    } while (select(s + 1, &readfds, NULL, NULL, &timeout) > 0
+        openssl_fdset(sock, &readfds);
+    } while (select(sock + 1, &readfds, NULL, NULL, &timeout) > 0
```

Comment:
How timeout is maintained?

Reply:
In this part of the code I only improved the variable name.
The question you pose is not related to the adaptations I made to this function, 
neither to the variable name nor to the changes in the `PROTO_CONNECT` case, which explicitly does not use a timeout, and I did not touch the timeout behavior here.